src/App.tsx

```diff
@@ -1,50 +1,64 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
 import './App.css';
-import Search from './components/Search';
-import People from './components/People/People';
-import PostService from './API/PostService';
+import Search from './components/Search/Search';
+import Products from './components/Products/Products';
+import { getProducts } from './API/api';
 import Loader from './components/UI/Loader/loader';
 import './styles/app.css';
-import ErrorBoundary from './components/errorBoundary/ErrorBoundary';
-export default class App extends React.Component {
-  state = {
-    people: [],
-    isDataLoading: false,
-    searchItem: '',
-  };
+import Pagination from './components/Pagination/Pagination';
+import { getCountPages } from './utils/countPages';
+import LimitItems from './components/LimitItems/LimitItems';
+
+export default function App() {
+  const [products, setProducts] = useState([]);
+  const [isDataLoading, setIsDataLoading] = useState(false);
+  const [searchItem, setSearchItem] = useState('');
+  const [page, setPage] = useState(0);
+  const [limitPages, setLimitPages] = useState(10);
+  const [totalCount, setTotalCount] = useState(0);
+  const [totalPages, setTotalPages] = useState(0);
 
-  async fetchPeople() {
-    this.setState({ isDataLoading: true });
+  const fetchData = async () => {
+    setIsDataLoading(true);
     try {
-      const response = await PostService.getAll();
-      this.setState({ people: response.data.results });
+      const response = await getProducts(limitPages, page);
+      setProducts(response.products);
+      setTotalCount(response.products.length);
+      setTotalPages(getCountPages(totalCount, limitPages));
+      console.log(response);
     } catch (error) {
       console.error('Error fetching posts:', error);
     }
-    this.setState({ isDataLoading: false });
-  }
-
-  componentDidMount() {
-    this.fetchPeople();
-  }
-
-  setSearchItem = (searchValue: string) => {
-    this.setState({ searchItem: searchValue });
+    setIsDataLoading(isDataLoading);
   };
 
-  render() {
-    return (
-      <div className="app-wrapper">
-        <Search setSearchItem={this.setSearchItem} />
-        {this.state.isDataLoading ? (
-          <Loader></Loader>
-        ) : (
-          <People
-            people={this.state.people}
-            searchItem={this.state.searchItem}
+  useEffect(() => {
+    fetchData();
+    setTotalPages(getCountPages(totalCount, limitPages));
+  }, [page, totalCount, limitPages]);
```

Comment:
Terrible! useEffect is triggered many times and calls itself. and setTotalPages is called several times.